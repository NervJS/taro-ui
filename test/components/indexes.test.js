import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtIndexes from '../../.temp/components/indexes/index'

describe('AtIndexes Snap', () => {
  it('render initial AtIndexes', () => {
    const component = renderToString(<AtIndexes />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIndexes -- props className', () => {
    const component = renderToString(<AtIndexes className='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIndexes -- props customStyle', () => {
    const component = renderToString(<AtIndexes customStyle='color:red;' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIndexes -- props animation', () => {
    const component = renderToString(<AtIndexes animation />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIndexes -- props topKey', () => {
    const component = renderToString(<AtIndexes topKey='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtIndexes -- props children', () => {
    const component = renderToString(<AtIndexes>children</AtIndexes>)
    expect(component).toMatchSnapshot()
  })

  it('render AtIndexes -- props list', () => {
    const list = [
      {
        title: 'A',
        key: 'A',
        items: [
          {
            name: '阿坝'
          },
          {
            name: '阿里'
          },
          {
            name: '阿拉善'
          },
          {
            name: '阿拉善'
          },
          {
            name: '阿拉善'
          },
          {
            name: '阿拉善'
          },
          {
            name: '阿拉善'
          }
        ]
      },
      {
        title: 'B',
        key: 'B',
        items: [
          {
            name: '百度'
          },
          {
            name: '白色'
          },
          {
            name: '百强'
          }
        ]
      }
    ]
    const component = renderToString(<AtIndexes list={list} />)
    expect(component).toMatchSnapshot()
  })
})
