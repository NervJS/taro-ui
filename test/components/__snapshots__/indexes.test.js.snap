// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtIndexes Snap render AtIndexes -- props animation 1`] = `"<div class=\\"at-indexes\\">undefined<div class=\\"at-indexes__menu\\"><div class=\\"at-indexes__menu-item\\">Top</div></div><div id=\\"indexes-list-AOTU2018\\" scrollY scrollWithAnimation scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-y at-indexes__body\\"><div id=\\"at-indexes__top\\" class=\\"at-indexes__content\\"></div></div></div>"`;

exports[`AtIndexes Snap render AtIndexes -- props children 1`] = `"<div class=\\"at-indexes\\">undefined<div class=\\"at-indexes__menu\\"><div class=\\"at-indexes__menu-item\\">Top</div></div><div id=\\"indexes-list-AOTU2018\\" scrollY scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-y at-indexes__body\\"><div id=\\"at-indexes__top\\" class=\\"at-indexes__content\\">children</div></div></div>"`;

exports[`AtIndexes Snap render AtIndexes -- props className 1`] = `"<div class=\\"at-indexes test\\">undefined<div class=\\"at-indexes__menu\\"><div class=\\"at-indexes__menu-item\\">Top</div></div><div id=\\"indexes-list-AOTU2018\\" scrollY scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-y at-indexes__body\\"><div id=\\"at-indexes__top\\" class=\\"at-indexes__content\\"></div></div></div>"`;

exports[`AtIndexes Snap render AtIndexes -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-indexes\\">undefined<div class=\\"at-indexes__menu\\"><div class=\\"at-indexes__menu-item\\">Top</div></div><div id=\\"indexes-list-AOTU2018\\" scrollY scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-y at-indexes__body\\"><div id=\\"at-indexes__top\\" class=\\"at-indexes__content\\"></div></div></div>"`;

exports[`AtIndexes Snap render AtIndexes -- props list 1`] = `"<div class=\\"at-indexes\\">undefined<div class=\\"at-indexes__menu\\"><div class=\\"at-indexes__menu-item\\">Top</div><div class=\\"at-indexes__menu-item\\">A</div><div class=\\"at-indexes__menu-item\\">B</div></div><div id=\\"indexes-list-AOTU2018\\" scrollY scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-y at-indexes__body\\"><div id=\\"at-indexes__top\\" class=\\"at-indexes__content\\"></div><div id=\\"at-indexes__list-A\\" class=\\"at-indexes__list\\"><div class=\\"at-indexes__list-title\\">A</div><div class=\\"at-list\\"><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">阿坝</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">阿里</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">阿拉善</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">阿拉善</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">阿拉善</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">阿拉善</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">阿拉善</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div></div></div><div id=\\"at-indexes__list-B\\" class=\\"at-indexes__list\\"><div class=\\"at-indexes__list-title\\">B</div><div class=\\"at-list\\"><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">百度</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">白色</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">百强</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefinedundefined</div></div></div></div></div></div></div>"`;

exports[`AtIndexes Snap render AtIndexes -- props topKey 1`] = `"<div class=\\"at-indexes\\">undefined<div class=\\"at-indexes__menu\\"><div class=\\"at-indexes__menu-item\\">test</div></div><div id=\\"indexes-list-AOTU2018\\" scrollY scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-y at-indexes__body\\"><div id=\\"at-indexes__top\\" class=\\"at-indexes__content\\"></div></div></div>"`;

exports[`AtIndexes Snap render initial AtIndexes 1`] = `"<div class=\\"at-indexes\\">undefined<div class=\\"at-indexes__menu\\"><div class=\\"at-indexes__menu-item\\">Top</div></div><div id=\\"indexes-list-AOTU2018\\" scrollY scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-y at-indexes__body\\"><div id=\\"at-indexes__top\\" class=\\"at-indexes__content\\"></div></div></div>"`;
